.changeset/rich-rivers-cheer.md

```diff
@@ -0,0 +1,18 @@
+---
+'@astrojs/telemetry': minor
+'astro': minor
+---
+
+Adds a configuration option to disable CLI telemetry for the project
+
+Astro collects optional anonymous telemetry to help us understand how people use the CLI. This change adds a new way to disable telemetry on a per-project basis. Currently you can disable telemetry globally per machine by running `astro telemetry disable` or by setting the environment variable `ASTRO_TELEMETRY_DISABLE`. This change adds a new configuration option to disable telemetry for a specific project, which applies to all users of the project.
+
+```js
+// astro.config.mjs
+import { defineConfig } from 'astro/config';
+
+export default defineConfig({
+  disableTelemetry: true
```

Comment:
I think it would be nicer to have `telemetry: boolean`, defaulting to true

Reply:
Because telemetry starts before the config?

Reply:
No, I mean that naming it like that might make users thing that `true` would override their global values, but actually `true` would mean "true unless there's a global override or environment variable to disable it".

Reply:
mmm i see. I think I'd still prefer `telemetry: boolean` but log a warning if there's a mismatch with a global override